src/celengine/object.h

```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+#include <string>
+#include <set>
+
+class UserCategory;
+
+class Object
+{
+    typedef std::set<UserCategory*> cat_set;
+
+    cat_set *m_cats;
```

Comment:
I'd make it not a pointer but just `CategorySet m_cats`. Of course if you have tests that prove that it's worth to have such optimization here, then I'm okay with pointer.

Reply:
I just realized, that this class will be instantiated by thausend or more times, so its size should be minimized. However, I didn't test how much memory take an empty set. This could depend on compiler or library implementation.